hise-modules/effects/list/convolution.md

```diff
@@ -18,17 +18,49 @@ You can load a pooled audio file into this module and it will use it as impulse
 
 > This module can also be used for cabinet simulation when a cabinet IR is used.
 
+### Loading a single impulse response called impulse.wav
 
-
-### Load AudioFiles per script
+The file impulse.wav should be placed in the AudioFiles folder. This script assumes there is a Convolution Reverb already added called `Convolution Reverb1`.
 
 ```javascript
 // Create a typed Audio Sample Script Reference with right click on the Convolution Module
 const var ConvolutionReverb1 = Synth.getAudioSampleProcessor("Convolution Reverb1");
 
-// load all Audio Files from the AudioFiles Folder into the Pool
+// Load all Audio Files from the AudioFiles Folder into the Pool
 Engine.loadAudioFilesIntoPool();
 
-// set a new file 
+// Load the impulse response. Use of {PROJECT_FOLDER} here automatically refers to the AudioFiles folder.
 ConvolutionReverb1.setFile("{PROJECT_FOLDER}impulse.wav");
 ```
+
+### Including several user-selectable impulse responses with a plugin
+Impulse Response files should be placed in the AudioFiles project directory. If [Embed Audio Files](/working-with-hise/settings/project.html#embed-audio-files) is enabled, they will be embeded into the compiled plugin.
+
+Call [`Engine.loadAudioFilesIntoPool()`](/scripting/scripting-api/engine/index.html#loadaudiofilesintopool) to ensure the files are accessible in the compiled plugin. This returns an array of paths in the `{PROJECT_FOLDER}filename.wav` format, which can be directly loaded into the Convolution Reverb.
+
+```javascript
+// Create a typed Audio Sample Script Reference with right click on the Convolution Module
+const var ConvolutionReverb1 = Synth.getAudioSampleProcessor("Convolution Reverb1");
+
+// Create a comboBox
+const var impulseSelector = Content.addComboBox("impulseSelector",0,0);
+
+// Clear the items in the combobox. Otherwise, the entire list will be appended each time the script is run.
+impulseSelector.set("items","");
+
+// Load all Audio Files from the AudioFiles Folder into the Pool and get the list of references
+const var impulses = Engine.loadAudioFilesIntoPool();
+
+// Add names of impulse responses to the comboBox
+for (impulse in impulses){ 
+  impulseSelector.addItem(impulse.replace(".wav","").replace("{PROJECT_FOLDER}","")); //Add just the name, removing the .wav extension and `{PROJECT_FOLDER}`
+}
+
+// Create custom callback for the comboBox
+inline function onimpulseSelectorControl(component, value)
+{
+  ConvolutionReverb1.setFile(impulses[value - 1]); //Load the selected IR. Note that the comboBox value is 1 indexed
```

Comment:
Do you get an error for a new combo box which will have a value of 0?

Reply:
Not sure I follow, that should only happen if there are no items in the Audio Files folder?

Reply:
When you make a combobox I believe its initial value is 0. And 0 - 1 = invalid array index. I haven't tested this to confirm that it's an issue though.

Reply:
I just tested it, a freshly created comboBox has a value of 1.0.
```
HiseSnippet 1100.3ocuVs0SibCE1SfYUI8h1sZ66V7Tnhxl.YCfPUkkbojVtDQX219DxYlSRrvi8HOd.hVw68w9Z+2z+C8OR+Gzd7LSHyDnT1n1NODEet9YeNmO6dZkGDEozDmxmOIDHNeha+IRy3liYbIoaKhym4dLKx.ZZpnClDxhh.ehiyReqUfyJKSR99iu4.lfI8fYhHj2o3dvQ7.tYlzd6+8bgnCyGNmGjy55620SIapDpXDOK4VkDx7tjMBNgYMqjK4PVzXhyW5V80M.X6sF56y1ZK+FMZr6N6Ve2ZrgUqyFTqwt02r91aty1CINOqsO2nz8MLCDgA8.k+j9iUWKSSv63Q7ABvtnFoOl4Twjli4B+dSObhHDmk6M6nZozipW5dL2mem7YGYOOQAclG4OzbJ8XPp1G.jbxAokSgzKb66o4glYZr34ic6JwJ3PFVaxCkTaIk9kRtMUnERyFArKgNZbwcdToQ0pqSweVauxu5UzlZ.wMkQMXB8ouI1mqn8YAgBflFO5YvPPCXi.8ZtYLUyGM1P8DbuKoJI0LFnX1tRIhMbb8wJ+XATFq7QF5ULcdkmAWA5A0neMMoobiQfIIgo46tMYkUyGvLmVEwaA.6oBFnNPcStTwCBiEQPeP.dXIAyyzyAlueyL6qr5bls55UWuZwn6EGYTATOlPL.6YoCwfY2n2kStTvk.cXrzKAjJ4bA0lXsRTA8HTIQHrNhPQLrV42WdETYjR.aDp4RSkT46U9VDByEkMh.SVjZlgkJ+coZs8xglSTF3TYkjrU91xz4UMb3CpKKTBP+fpsy25GywJx3fAfN+d0ZH1OWbH4YOsgDuzpWNCUxtRt4zPHacGkv217a++8GoHYke6vUFBQSMIiVOOazZZaAg6am2l6rkjrQxSJRda2VLCaZHwniYLDzFtci4zBtBYHSmdWwsEDcoQElXaVe.ww8CB.2fN3XS8jo+IfcmLx8oLPxJ6DHyTjAyxsmo.qXEnMrTCxHtYR9s4G.sV0GkV6oBwW31ia7F+vXrzCfQrB7eAFytL3ScaObHVAlAvkc67iKJye0mNy+mOM+43.Sx+W39.rhE.TK8jhWg+S+1O.lhhHjiPzJ8JbP1Msm6HPNxLNeWPFt5JCiM4k+1H3.jKZjVEK8OeLRZ5mOd8zfOHXExwg7l4hwO+662BI84xQ4soSmyylbxD8x86vEYuW3qbeeuyN86Z277K5b5QsZe1s0nGhjhQzZah2XI8wyxDAabM6JRfcSWXd4WuYIhPoBwxk1LUmUPao+T8NK8uzk4jyTXcRN5XlQyso+j3f936f7.rgRJAQjMkkrTToqqNEQ8Aoexh+D+xTVyt1ISYsoJ+eIGALOs5BuTxSau5GkHA22xjpxJ3yIw0zZyyThm63iltvyqXntmiatnNt0h5X8E0wWunN1XQcb6E0wc9mcz9l12DiOyIkskPNtW6ja1bbZKYXWdBmB4uLj338M
```

Reply:
Ah cool, no problem then